### PR TITLE
fix #2166

### LINKED ISSUE #2166: Use a FS cache for Jinja to speed it up a bit
http://jinja.pocoo.org/docs/dev/api/#bytecode-cache

### PR DESCRIPTION
**Pull request by ralsina**
Use a FS cache for jinja bytecode like we do for Mako.

### CODE REVIEW

File: nikola/plugins/template/jinja.py [hunk 1]
```diff
@@ -52,7 +52,10 @@ def __init__(self):
         """Initialize Jinja2 environment with extended set of filters."""
         if jinja2 is None:
             return
-        self.lookup = jinja2.Environment()
+        # Not ideal because it's not using our cache structure
```
Comment (Kwpolska) [inline nikola/plugins/template/jinja.py]: Where is it held then? If this is the case, couldn’t we handle this in `create_lookup` or `set_site`?

Comment (ralsina) [inline nikola/plugins/template/jinja.py]: According to the jinja docs "system temp folder". It looks like it needs to be set on env creation. Maybe we can delay that until set_site?

Comment (ralsina) [inline nikola/plugins/template/jinja.py]: I'll see if I can fix this.

Comment (ralsina) [inline nikola/plugins/template/jinja.py]: Done I think.

**Comment by Kwpolska** [comment]
Merge if this does not break anything (test it yourself)

**Pull request merged**